lib/ansible/playbook/collectionsearch.py

```diff
@@ -19,12 +19,14 @@ def _ensure_default_collection(collection_list=None):
     if collection_list is None:
         collection_list = []
 
-    if default_collection:  # FIXME: exclude role tasks?
-        if isinstance(collection_list, string_types):
-            collection_list = [collection_list]
+    # We allow collections to be either a string or a list, so make sure
+    # we have a list going forward.
+    if isinstance(collection_list, string_types):
```

Comment:
Should we insert an `elif not isinstance(collection_list, (None, list)):` and fail?  As it's written, someone could supply all sorts of other incorrect data types.

Reply:
Good idea. If I make it a dict, we get the same traceback about not being able to call append(). I'll make that change.

Reply:
instead of custom code, why just not force validation? this would handle 'non lists' already correctly and since 'collections' is static, we don't  have to worry about templating

Reply:
@bcoca If we assume CollectionSearch is always mixed in with Base (seems like it is?), then we can call self.get_validated_value() early on. I'm unclear on if the mixin is a safe assumption going forward though.

Reply:
i really want to redesign it so 'get_validated_value' is the normal 'getter' and that 'get_unvalidated' becomes the method for when we need 'direct' access.